### ISSUE TITLE
Dataset __getitem__ is wrong

### ISSUE THREAD

**Issue by miquelferrarons** [harveyslash/Facial-Similarity-with-Siamese-Networks-in-Pytorch#7]
This part is working with the faces dataset only because the number of classes is big enough, but it will not generalize well to dataset where the number of classes is small (e.g. 2).

Basically, the else branch should also have a while True: with the opposite logic. Since you don't have that, the else branch may indeed choose an img1_tuple of the same class of img1_tuple, with a probability of 1/n_classes. So, if you have 40 classes, it's very unlikely that that will happen, thus your sample works fine, but if you have 2 classes that will happen 50% of the time, which would mean that instead of using 50%-50% you'd be using 75%-25%.

   ```
        should_get_same_class = random.randint(0, 1)
        if should_get_same_class:
            while True:
                # keep looping till the same class image is found
                img1_tuple = random.choice(self.imageFolderDataset.imgs)
                if img0_tuple[1] == img1_tuple[1]:
                    break
        else:
            img1_tuple = random.choice(self.imageFolderDataset.imgs)
```

I know it's only an example, but that part tricked me for a while, so I thought it was worth mentioning.

**Comment by harveyslash** [comment]
I see what you mean. 
Thanks for pointing it out! 
If you would like , you can send a pull request with the fix, and I can merge it.

**Comment by harveyslash** [comment]
fixed in 3f10b43

**Comment by swg209** [comment]
I got a question , what if I have 751 classes,  could I use the dataloader way  for my siamese network?
